Added guidance on how to add snowflake acct field

mirnawong1 opened the pull request:
Added guidance below based on users asking and having an issue on connecting. usually they include the snowflakecomputing.com. Please can someone verify this is correct guidance? thank you

Note: When connecting to Snowflake, pay close attention to the relevant parts of your Account URL when filling in the "Account" field.  For example, below you can see guidance on what to enter in the "Account" field in dbt Cloud or your profiles.yml file for dbt Core.

• If the URL for your Snowflake account is abc12345.eu-central-1.snowflakecomputing.com, then you should enter abc12345.eu-central-1. Basically, remove the snowflakecomputing.com portion of your URL

## Description & motivation
<!---
Describe your changes, and why you're making them. Is this linked to an open
issue, a pull request on dbt core, etc?
-->

## To-do before merge
<!---
(Optional -- remove this section if not needed)
Include any notes about things that need to happen before this PR is merged, e.g.:
- [ ] Change the base branch
- [ ] Ensure PR #56 is merged
-->

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

mirnawong1 committed:
Added guidance on how to add snowflake acct field

Added guidance below based on users asking and having an issue on connecting. usually they include the snowflakecomputing.com. Please can someone verify this is correct guidance? thank you

Note: When connecting to Snowflake, pay close attention to the relevant parts of your Account URL when filling in the "Account" field.  For example, below you can see guidance on what to enter in the "Account" field in dbt Cloud or your profiles.yml file for dbt Core.

• If the URL for your Snowflake account is abc12345.eu-central-1.snowflakecomputing.com, then you should enter abc12345.eu-central-1. Basically, remove the snowflakecomputing.com portion of your URL

mirnawong1 commented:
Added guidance below based on users asking and having an issue on connecting. usually they include the snowflakecomputing.com. Please can someone verify this is correct guidance? thank you

mirnawong1 committed:
Update connecting-your-database.md

refined edits to use white checkmark and red cross

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

dbeatty10 commented:
@mirnawong1 It looks like we have similar Snowflake account information listed here:
https://docs.getdbt.com/guides/getting-started/getting-set-up/setting-up-snowflake#connect-dbt-cloud-to-snowflake

Is there any way to keep both locations in-sync somehow?

## Some research
I haven't verified any of the following personally, so your mileage may vary.

It sounds like Snowflake has two formats:
1. [Format 1 (Preferred): Account Name in Your Organization](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#format-1-preferred-account-name-in-your-organization)
2. [Format 2 (Legacy): Account Locator in a Region](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#format-2-legacy-account-locator-in-a-region)

It _might_ be that only Format 2 works for dbt + Snowflake currently. If so, that means dbt-snowflake would need a format like [this](https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#non-vps-account-locator-formats-by-cloud-platform-and-region).

## Ideas for querying the values
Unverified(!) monstrosities like the following might work for querying the relevant Snowflake Account Locator (assuming the user has logged into the Snowflake Web UI:

### Option 1: One query with substrings
```sql
select lower(current_account() || '.' || replace(substring(current_region(), position('_', current_region())+1), '_', '-') || '.' || left(current_region(), position('_', current_region())-1)) as account_locator;
```

### Option 2: Three queries with `result_scan`
```sql
show regions;
set show_regions = last_query_id();
select
  lower(current_account()) || '.' || "region" || '.' || "cloud" as account_locator
from table(result_scan($show_regions))
where "snowflake_region" = current_region()
;
```

Here are the relevant Snowflake docs for each of the functions from the monstrosities:
- [current_region](https://docs.snowflake.com/en/sql-reference/functions/current_region.html)
- [current_account](https://docs.snowflake.com/en/sql-reference/functions/current_account.html)
- [position](https://docs.snowflake.com/en/sql-reference/functions/position.html)
- [result_scan](https://docs.snowflake.com/en/sql-reference/functions/result_scan.html)
- [substring](https://docs.snowflake.com/en/sql-reference/functions/substr.html)
- [replace](https://docs.snowflake.com/en/sql-reference/functions/replace.html)
- [lower](https://docs.snowflake.com/en/sql-reference/functions/lower.html)

mirnawong1 commented:
thank you @dbeatty10 that's really helpful and i will check with the team to see if there is a way to keep them in sync. 

to clarify, do you mean if something changes on this document, it should be reflected [here](https://docs.getdbt.com/guides/getting-started/getting-set-up/setting-up-snowflake#connect-dbt-cloud-to-snowflake) as well?

i am proposing making the below changes to [this](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database#connecting-to-snowflake) document so it's simpler and clearer:

`:white_check_mark: Use: `db5261993`,`db5261993.east-us-2.azure` <br> :x: Avoid: `db5261993.eu-central-1.snowflakecomputing.com` |`

dbeatty10 commented:
Your proposed changes to make it simpler and clearer look great and make sense. 🤩

So my ideas & suggestions aren't directly relevant to your edits!

My ideas/suggestions are basically two-fold:
1. [Getting Started with dbt Cloud → Getting set up → Set up and connect Snowflake](https://docs.getdbt.com/guides/getting-started/getting-set-up/setting-up-snowflake#connect-dbt-cloud-to-snowflake) should match [Configuring dbt Cloud → Connecting your database](https://docs.getdbt.com/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database#connecting-to-snowflake).
2. [Profiles (CLI only) → Snowflake Profile → Account](https://docs.getdbt.com/reference/warehouse-profiles/snowflake-profile#account) could contain a verified version of one or both of the example queries prototyped above.
  - It could also contain your emoji hints:
      - :white_check_mark: `db5261993`,`db5261993.east-us-2.azure` <br> :x: `db5261993.eu-central-1.snowflakecomputing.com`

mirnawong1 committed:
Update connecting-your-database.md

added snippet src file name per instructions here: https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md#reusing-snippets-of-content

snippet pr is here: https://github.com/dbt-labs/docs.getdbt.com/pull/1923

mirnawong1 commented:
in order to sync both pages, i've added src snippet per instructions here: https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md#reusing-snippets-of-content

i do need to have pr https://github.com/dbt-labs/docs.getdbt.com/pull/1923 approved first and this is also included in the open pr https://github.com/dbt-labs/docs.getdbt.com/pull/1924 as well

runleonarun committed:
Merge branch 'current' into mirnawong1-patch-5

runleonarun commented on website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md:
```diff
@@ -53,7 +53,7 @@ The following fields are required when creating a Snowflake connection:
 
 | Field | Description | Examples |
 | ----- | ----------- | -------- |
```
```suggestion
| ----- | ----------- | ---------------- |
```

runleonarun committed:
Update website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md

runleonarun commented on website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-your-database.md:
```diff
@@ -52,8 +52,8 @@ Once the connection is saved, a public key will be generated and displayed for t
 | Field | Description | Examples |
-| ----- | ----------- | -------- |
-| Account | The Snowflake account to connect to. Take a look [here](snowflake-profile#account) to determine what the account field should look like based on your region.| `db5261993`,`db5261993.east-us-2.azure` |
+| ----- | ----------- | ---------------- |
```
@mirnawong1 this is working now! I know there's a way to make the last column wider in markdown, but I can't remember how. If you want to add that, then maybe you can look into it? Otherwise this looks great!

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Update connecting-your-database.md

added &nbsp; &nbsp; to widen last column

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Update connecting-your-database.md

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5

dbeatty10 reviewed (approved):
Looks awesome @mirnawong1

mirnawong1 commented:
yay thank you sooo much for your help @dbeatty10 ! you da best!

mirnawong1 committed:
Merge branch 'current' into mirnawong1-patch-5
